### PR TITLE
feat: always send http request "content-type" header

### PR DESCRIPTION
**Pull request by danbosnichill**
I don't know why we didn't always send through "content-type" header.

This PR also tries to fix GitHub Action configs.  "macos-latest" uses a version of macos that fails the GHA.

TESTING=Iterated on GHA.  I'm iterating on the test.

### CODE REVIEW

**Review by prm-james-hill — APPROVED** [review]

File: Sources/PromotedCore/NetworkConnection.swift [hunk 1]
```diff
@@ -85,18 +85,26 @@ public extension NetworkConnection {
...
   }
+
+  func contentType(
+    clientConfig: ClientConfig
+  ) -> String {
+    switch clientConfig.metricsLoggingWireFormat {
+    case .binary:
+      return "application/protobuf"
+    case .json:
+      return "application/json"
```
Comment (prm-xingcan) [inline Sources/PromotedCore/NetworkConnection.swift:105]: Didn't have too much context on this, but I assume JSON and proto binary are the only supported types here.

**Review by prm-xingcan — APPROVED** [review]

**Comment by prm-xingcan** [comment]
Might need to fix the tests

File: Sources/PromotedCore/NetworkConnection.swift (hunk 1 above)
Comment (danbosnichill) [reply Sources/PromotedCore/NetworkConnection.swift:105]: Yes.

**Comment by danbosnichill** [comment]
> Might need to fix the tests
Yea, I'm still iterating on it.  It works locally.

**Pull request merged**